### PR TITLE
Drop e2etest directory

### LINKED ISSUE #191: Add Romana to the e2e test matrix
I want e2e tests that verify against a matrix of the popular network addons for k8s: Flannel, Calico, Romana, and Weave (and kube-router if/when we can get compatibility).

This is currently a little hard with Romana, because the manifests require some customization when using a non-default service CIDR, and those modifications are described in long-form prose rather than something I can search/replace like with Calico and Flannel.

That said, I still want to test with Romana, not least because it has custom integration instructions that need to be exercised. Filing this bug to remind myself to make that work, once I have some of the rest of the test matrix up and working.

### LINKED ISSUE #192: Add e2e tests running on Virtuakube
Should set up full e2e regression tests that spin up a full cluster on GCP, using various network providers and cluster settings, and runs a series of basic functionality tests.

I finally have a working Terraform config that can stand up an IPv4 cluster using {flannel, calico, romana, weave} for networking, with a working layer-2 network that can be used to test both BGP and ARP mode.

Currently, IPv6 is still a bit borked because of https://github.com/kubernetes/kubernetes/pull/58769 , so we can't test NDP just yet (plus, calico's config would need a bunch of tweaking). But, once the framework is in place, we can go there.

### LINKED ISSUE #629: e2etest/main_test.go still uses make.go
Hi!

I'm learning how MetalLB development works so I'm trying to run the e2etests and stubbed my toe.  Using MetalLB a9a6f319 running the tests looks like:
```
$ go test ./e2etest
stat make.go: no such file or directory
Failed to build MetalLB images: exit status 1FAIL	go.universe.tf/metallb/e2etest	0.007s
FAIL
```
The problem seems to be e2etest/main_test.go.  It still depends on make.go but that was replaced by Python invoke.  Maybe I missed something and am running the test incorrectly, corrections appreciated!

I hacked main_test.go to work with inv instead of make.go so please LMK if you'd like me to submit a PR to fix it, or if that file is no longer used I can submit a PR to delete it so other newbies don't stub their toes on it.

Thanks!

### PR DESCRIPTION
**Pull request by russellb**
These tests no longer work.  They rely on the now archived virtuakube,
which doesn't build against kube 1.19, which ends up blocking fixing
metallb for kube 1.19.  See #723 for discussion of where it gets in the
way.

At this point we should just drop this directory and re-introduce
e2etest code once someone is able to work on it.  We can always pull
this back out of git history if it helps.

The commit contents are the result of roughly:

 - git rm -r e2etest
 - go mod tidy
 - git add go.mod go.sum

Closes #192 
Closes #629
Closes #639

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -22,11 +21,9 @@ require (
 	github.com/osrg/gobgp v2.0.0+incompatible
 	github.com/prometheus/client_golang v1.0.0
 	github.com/satori/go.uuid v1.2.0 // indirect
-	github.com/spf13/cobra v0.0.5
-	github.com/spf13/pflag v1.0.3
+	github.com/spf13/viper v1.3.2 // indirect
```
Comment (rata) [inline go.mod:24]: Just curious, what did you run exactly to update this?

**Review by rata — APPROVED** [review]
LGTM, thanks!

File: go.mod (hunk 1 above)
Comment (russellb) [reply go.mod:24]: `go mod tidy`

**Review by daxmc99 — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (rata) [reply go.mod:24]: Oh, perfect. Thanks! :)

**Comment by champtar** [comment]
@russellb can you fix the conflict ?

**Comment by rata** [comment]
This removes the mirror-server yaml, that is used in `inv dev-env`, so it breaks that :-(

**Comment by rata** [comment]
I plan to merge this fixing the conflict, as it makes it way easier to upgrade client-go. Will update soon

**Pull request closed without merging**